container/container_unix.go

```diff
@@ -280,18 +281,50 @@ func (container *Container) UnmountSecrets() error {
 
 // UpdateContainer updates configuration of a container.
 func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfig) error {
+	sysInfo := sysinfo.New(true)
+
 	container.Lock()
 	defer container.Unlock()
 
 	// update resources of container
 	resources := hostConfig.Resources
 	cResources := &container.HostConfig.Resources
+
+	// validate NanoCPUs, CPUPeriod, and CPUQuota
+	if resources.NanoCPUs > 0 && resources.CPUPeriod > 0 {
```

Comment:
I think most of these validation steps are already done in  [`verifyContainerResources ()`](https://github.com/docker/docker/blob/f7819fcb25aaf5b08202db275847c825e7b09999/daemon/daemon_unix.go#L368-L422); which is called by

`daemon.ContainerUpdate()`
   -> `daemon.verifyContainerSettings()`
   -> `verifyPlatformContainerSettings()`
   -> `verifyContainerResources()`

So already triggered [_before_ this verification is called](https://github.com/docker/docker/blob/737b5b1781da9d7f286bb2425720add0ff69e3e3/daemon/update.go#L13-L20)